scripts/build_example

```diff
@@ -55,7 +55,7 @@ fi
 
 readonly EXAMPLE_BIN_DIR="${PWD}/examples/${EXAMPLE}/bin"
 rm -rf "${EXAMPLE_BIN_DIR:?}"
-mkdir --parents "${EXAMPLE_BIN_DIR}"
+mkdir -p "${EXAMPLE_BIN_DIR}"
```

Comment:
What version of `mkdir` does macos use that does not recognise those long flags? FYI I think we use them in a bunch of other places too, so maybe those need to be fixed as well?

Reply:
macOS uses ancient versions of a bunch of tools, sadly. 

I've updated the PR with a few other things (including the other uses of `--parents`, thanks), PTAL.